File: golang/链表/移除链表元素.go
package main

func removeElement(head *ListNode, val int) {
	dummy := &ListNode{Next: head}

	for cur := dummy; cur != nil && cur.Next != nil; {
		if cur.Next.Val == val {
			cur.Next = cur.Next.Next
		} else {
			cur = cur.Next
		}
	}
}

func removeElement1(head *ListNode, target int) *ListNode {
	dummy := &ListNode{
		Val:  0,
		Next: head,
	}

	cur := dummy
	for cur != nil && cur.Next != nil {
		if cur.Next.Val == target {
			cur.Next = cur.Next.Next
		} else {
			cur = cur.Next
		}
	}

	return dummy.Next
}

File: python/链表/移除链表元素.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def removeElements(head, val):
    dummy = ListNode(0)
    dummy.next = head
    cur = dummy
    while cur.next:
        if cur.next.val == val:
            cur.next = cur.next.next
        else:
            cur = cur.next
    return dummy.next
